src/app/shared/background.model.ts

```diff
@@ -1,15 +1,44 @@
-import { Die, } from './die.model';
+import { Die } from './die.model';
 
 export class CharacterBackground {
-    constructor(
-    public number: number,
-    public name: string,
-    public backgroundText: string,
-    public qualities: string[],
-    public principleCategory: string,
-    public powerSourceDice: Die[],
-    public diceText: string,
-    public pageNumber: number,
-    public diceToAssign: Die[] = [],
-    ) {}
+    public number: number;
+    public name: string;
+    public backgroundText: string;
+    public qualities: string[];
+    public principleCategory: string;
+    public powerSourceDice: Die[] = [];
+    public diceText: string;
+    public pageNumber: number;
+    public diceToAssign: Die[] = [];
+
+    constructor(backgroundData: {
```

Comment:
I like how you have updated the constructor to take in an object.  In order to make this efficient, there are some things I would consider doing to better facilitate this.  As of now, the object it accepts is defined here in the class.  It would be better if it was an interface, so that we could use it in multiple places.  This would also allow us to have consistency between the properties on the class and its constructor object.
```
// i.e.
export class CharacterBackground implements ICharacterBackground {
...
    constructor(backgroundData: ICharacterBackground)
...
```

Reply:
I will definitely look at how to do this properly and see what I can come up with